ghcide/src/Development/IDE/Plugin/TypeLenses.hs

```diff
@@ -294,3 +324,116 @@ pprPatSynTypeWithoutForalls p = pprPatSynType pWithoutTypeVariables
     builder = patSynBuilder p
     field_labels = patSynFieldLabels p
     orig_args' = map scaledThing orig_args
+
+-- --------------------------------------------------------------------------------
+
+-- | A binding expression with its id(s) and location.
+data WhereBinding = WhereBinding
+    { bindingId  :: [Id]
+    -- ^ There may multiple ids for one expression.
+    -- e.g. @(a,b) = (1,True)@
+    , bindingLoc :: SrcSpan
+    -- ^ Location for the whole binding.
+    -- Here we use the this to render the type signature at the proper place.
+    --
+    -- Example: For @(a,b) = (1,True)@, it will print the signature after the
+    -- open parenthesis instead of the above of the whole expression.
```

Comment:
The nice thing about this is that you then can have one lens per id, which I think simplifies the whole `WhereBindings` thing significantly. It's also nicer output: singatures on separate lines, rather than inline, and you can only specify a type for a subset of the ids if you only care about some of them.

Reply:
Currently, it behaves as you described.

Reply:
Oh, I misunderstood the comment! The comment is saying that it *would* go in the wrong place if you didn't use the span for the whole binding. I get it now!